httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java

```diff
@@ -541,6 +560,18 @@ public URIBuilder setPathSegments(final List<String> pathSegments) {
         return this;
     }
 
+    /**
+     * Appends segments to URI path. The value is expected to be unescaped and may contain non ASCII characters.
+     *
+     * @return this.
+     */
+    public URIBuilder appendPathSegments(final List<String> pathSegments) {
+        final List<String> segments = new ArrayList<>(getPathSegments());
```

Comment:
@larshelge Same here.

Reply:
Agreed @ok2c. I wanted to avoid if-statements but we can favor less heap usage here.

Note that the `new ArrayList<>(getPathSegments())` wrapper list is necessary as `getPathSegments` might return an immutable list which cannot be added to later.

I have made the updates now.